helm/v1alpha1/types_helm.go

```diff
@@ -0,0 +1,108 @@
+package v1alpha1
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:plural=helmchartrepositories
+
+// HelmChartRepository holds cluster-wide configuration for proxied Helm chart repository
+type HelmChartRepository struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec HelmChartRepositorySpec `json:"spec"`
+
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status HelmChartRepositoryStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type HelmChartRepositoryList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []HelmChartRepository `json:"items"`
+}
+
+// Helm chart repository exposed within the cluster
+type HelmChartRepositorySpec struct {
```

Comment:
Modelled according to Helm [chartrepo.Entry](https://github.com/helm/helm/blob/master/pkg/repo/chartrepo.go#L42)

Reply:
> Modelled according to Helm [chartrepo.Entry](https://github.com/helm/helm/blob/master/pkg/repo/chartrepo.go#L42)

model based on https://github.com/kubernetes/api/blob/master/admissionregistration/v1/types.go#L488 instead.  A separate struct with connection information.  I'll throw out

```go
type ClientConfig struct{
  URL string
  // empty means system roots
  CA configv1.ConfigMapNameReference
  // empty means no client cert.  tls.crt for cert, tls.key for key.  this matches the kube secret type
  ClientCert configv1.SecretNameReference
  // empty means no token.  key token (congruent to serviceaccount tokens) is used.
  Token configv1.SecretNameReference
}
```

Reply:
Done.